src/System.Data.Common/tests/System/Data/DataColumnTest.cs

```diff
@@ -361,280 +306,269 @@ public void ExpressionSubstringlimits()
+            DataColumn c = new DataColumn("name");
+            t.Columns.Add(c);
+            c = new DataColumn("age");
+            c.DataType = typeof(int);
+            t.Columns.Add(c);
+            c = new DataColumn("id");
+            c.Expression = "substring (name, 1, 3) + len (name) + age";
+            t.Columns.Add(c);
 
-            DataSet Set = new DataSet("TestSet");
-            Set.Tables.Add(T);
+            DataSet set = new DataSet("TestSet");
+            set.Tables.Add(t);
 
-            DataRow Row = null;
+            DataRow row = null;
             for (int i = 0; i < 100; i++)
             {
-                Row = T.NewRow();
-                Row[0] = "human" + i;
-                Row[1] = i;
-                T.Rows.Add(Row);
+                row = t.NewRow();
+                row[0] = "human" + i;
+                row[1] = i;
+                t.Rows.Add(row);
             }
 
-            Row = T.NewRow();
-            Row[0] = "h*an";
-            Row[1] = DBNull.Value;
-            T.Rows.Add(Row);
+            row = t.NewRow();
+            row[0] = "h*an";
+            row[1] = DBNull.Value;
+            t.Rows.Add(row);
 
-            Assert.Equal("hum710", T.Rows[10][2]);
-            Assert.Equal("hum64", T.Rows[4][2]);
-            C = T.Columns[2];
-            C.Expression = "isnull (age, 'succ[[]]ess')";
-            Assert.Equal("succ[[]]ess", T.Rows[100][2]);
+            Assert.Equal("hum710", t.Rows[10][2]);
+            Assert.Equal("hum64", t.Rows[4][2]);
+            c = t.Columns[2];
+            c.Expression = "isnull (age, 'succ[[]]ess')";
+            Assert.Equal("succ[[]]ess", t.Rows[100][2]);
 
-            C.Expression = "iif (age = 24, 'hurrey', 'boo')";
-            Assert.Equal("boo", T.Rows[50][2]);
-            Assert.Equal("hurrey", T.Rows[24][2]);
+            c.Expression = "iif (age = 24, 'hurrey', 'boo')";
+            Assert.Equal("boo", t.Rows[50][2]);
+            Assert.Equal("hurrey", t.Rows[24][2]);
 
-            C.Expression = "convert (age, 'System.Boolean')";
-            Assert.Equal(bool.TrueString, T.Rows[50][2]);
-            Assert.Equal(bool.FalseString, T.Rows[0][2]);
+            c.Expression = "convert (age, 'System.Boolean')";
+            Assert.Equal(bool.TrueString, t.Rows[50][2]);
+            Assert.Equal(bool.FalseString, t.Rows[0][2]);
 
             //
             // Exceptions
             //
 
-            try
-            {
-                // The expression contains undefined function call iff().
-                C.Expression = "iff (age = 24, 'hurrey', 'boo')";
-                Assert.False(true);
-            }
-            catch (EvaluateException)
-            {
-            }
-            catch (SyntaxErrorException)
-            {
-            }
-
+            // TODO: which exception is it exactly?
+            Assert.ThrowsAny<InvalidExpressionException>(() => c.Expression = "iff (age = 24, 'hurrey', 'boo')");
```

Comment:
catching EvaluateException and SyntaxErrorException simultaneously, which are a derived type of InvalidExpressionException. What is it supposed to throw exactly?